src/CompanyService.Data/OfficeRepository.cs

```diff
@@ -1,70 +1,86 @@
-﻿using LT.DigitalOffice.CompanyService.Data.Interfaces;
-using LT.DigitalOffice.CompanyService.Data.Provider;
-using LT.DigitalOffice.CompanyService.Models.Db;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using LT.DigitalOffice.CompanyService.Data.Interfaces;
+using LT.DigitalOffice.CompanyService.Data.Provider;
+using LT.DigitalOffice.CompanyService.Models.Db;
+using LT.DigitalOffice.CompanyService.Models.Dto.Requests.Filters;
+using LT.DigitalOffice.Kernel.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.JsonPatch;
 
 namespace LT.DigitalOffice.CompanyService.Data
 {
   public class OfficeRepository : IOfficeRepository
+  {
+    private readonly IDataProvider _provider;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public OfficeRepository(
+      IDataProvider provider,
+      IHttpContextAccessor httpContextAccessor)
     {
-        private readonly IDataProvider _provider;
+      _httpContextAccessor = httpContextAccessor;
+      _provider = provider;
+    }
 
-        public OfficeRepository(IDataProvider provider)
-        {
-            _provider = provider;
-        }
+    public void Add(DbOffice office)
```

Comment:
Can you make this method bool? And then remove throw exception.

Reply:
done